src/server/server/dal.py

```diff
@@ -223,6 +230,11 @@ def _to_db_doc(doc):
         feature_vector=_to_db_feature_vector(doc.feature_vector))
 
 
+def get_doc_by_urlsafe_key(key):
+    db_key = ndb.Key(urlsafe=key)
+    return _to_doc(db_key.get())
+
```

Comment:
Sorry, something i missed the last time I think : this function doesn't seem to be unit tested